pkg/nsx/services/firewall.go

```diff
@@ -432,60 +713,108 @@ func (service *SecurityPolicyService) buildRuleSrcGroup(obj *v1alpha1.SecurityPo
 		DisplayName: &ruleSrcGroupName,
 		Tags:        peerTags,
 	}
+
+	ruleSrcGroupCount := 0
+	var err error = nil
 	for i, peer := range sources {
-		service.updatePeerExpressions(obj, &peer, &ruleSrcGroup, i)
+		ruleSrcGroupCount += service.updatePeerExpressions(obj, &peer, &ruleSrcGroup, i)
 	}
+	if ruleSrcGroupCount > MaxGroupExpressions {
+		errorMsg := fmt.Sprintf("Total number of rule source group expressions:%d exceed NSX limit of %d", ruleSrcGroupCount, MaxGroupExpressions)
+		err = errors.New(errorMsg)
+		log.Error(err, "buildRuleSrcGroup")
```

Comment:
`buildRuleSrcGroup`: do not just add function name, use readable message

Reply:
Sure.